internal/spec/spec.go

```diff
@@ -41,14 +46,50 @@ func newExtraSpecsFromBootstrapData(data params.BootstrapInstance) (*extraSpecs,
 		}
 	}
 
+	if err := spec.Validate(); err != nil {
+		return nil, fmt.Errorf("failed to validate extra specs: %w", err)
+	}
+
 	return spec, nil
 }
 
+func (e *extraSpecs) Validate() error {
+	if len(e.CustomLabels) > 61 {
+		return fmt.Errorf("custom labels cannot exceed 64 items")
```

Comment:
just a nit. We can say: "custom labels cannot exceed 61 items". Because the user can only set that many. We set the other 3.